pkg/reconcilermanager/controllers/garbage_collector.go

```diff
@@ -171,31 +170,56 @@ func (r *RepoSyncReconciler) deleteRoleBinding(ctx context.Context, reconcilerRe
 	return nil
 }
 
-func (r *RepoSyncReconciler) deleteHelmValuesFileRefCopies(ctx context.Context, rs *v1beta1.RepoSync) error {
-	if rs == nil || rs.Spec.SourceType != string(v1beta1.HelmSource) || rs.Spec.Helm == nil || len(rs.Spec.Helm.ValuesFileRefs) == 0 {
-		return nil
+// deleteHelmConfigMapCopies deletes helm values file ConfigMap copies,
+// except those specified in the cmNamesToKeep set.
+func (r *RepoSyncReconciler) deleteHelmConfigMapCopies(ctx context.Context, rsRef types.NamespacedName, cmNamesToKeep map[string]struct{}) error {
+	// cleanup old copied ConfigMaps that we don't need anymore, e.g. if the list in valuesFileRefs changes
+	// TODO: Eventually remove the annotations and use the labels for list filtering, to optimize cleanup.
+	// We can't remove the annotations until v1.16.0 is no longer supported.
```

Comment:
Why is this? Would removal of the annotations be considered a breaking change?

Reply:
The problem is mostly around garbage collection. We need to make sure any CMs made with v1.16.0 are deleted if the config changes, namely between v1.16.0 and v1.16.1+. We could stop adding the annotation, but it's not really hurting anything, and might help if there's a downgrade. Plus, not everyone upgrades every revision. So by v1.19.0 (or whatever the +3 version is after synchronizing with GKE versions) we can drop the annotation and stop depending on it for GC.